pkg/kube/qdr/tls_handler.go

```diff
@@ -0,0 +1,206 @@
+	ExistsSslProfile(string) (bool, error)
+	RemoveSslProfile(string) error
+}
+
+type TlsServiceSupport struct {
+	Address       string
+	Credentials   string
+	CertAuthority string
+}
+
+func EnableTlsSupport(support TlsServiceSupport, tlsManager TlsManagerInterface) error {
+
+	if support.Credentials != "" {
+		if isGeneratedBySkupper(support.Credentials) {
+			// If the requested certificate is one generated by skupper it can be generated in other sites as well
+			_, err := tlsManager.GetSecret(support.Credentials)
+			if err != nil {
+				serviceSecret, err := generateNewSecret(support, tlsManager)
+				if err != nil {
+					return err
+				}
+
+				return includeSslProfile(qdr.SslProfile{Name: serviceSecret.Name}, tlsManager)
+			}
+		} else {
+			_, err := tlsManager.GetSecret(support.Credentials)
+			if err != nil {
+				return fmt.Errorf("The secret %s for address %s is missing", support.Credentials, support.Address)
+			}
+
+			err = checkAndIncludeSslProfile(qdr.SslProfile{Name: support.Credentials}, tlsManager)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if support.CertAuthority != "" && support.CertAuthority != types.ServiceClientSecret {
+		_, err := tlsManager.GetSecret(support.CertAuthority)
+		if err != nil {
+			return fmt.Errorf("The secret %s for address %s is missing", support.CertAuthority, support.Address)
+		}
+
+		sslProfile := qdr.SslProfile{
+			Name:       support.CertAuthority,
+			CaCertFile: fmt.Sprintf("/etc/skupper-router-certs/%s/ca.crt", support.CertAuthority),
+		}
+
+		err = checkAndIncludeSslProfile(sslProfile, tlsManager)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func DisableTlsSupport(credentials string, tlsManager TlsManagerInterface) error {
+	if len(credentials) > 0 {
+		if isGeneratedBySkupper(credentials) {
+			err := tlsManager.RemoveSslProfile(credentials)
+			if err != nil {
+				return err
+			}
+
+			_, err = tlsManager.GetSecret(credentials)
+			if err == nil {
+				err = tlsManager.DeleteSecret(credentials)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func CheckBindingSecrets(services map[string]*service.ServiceBindings, namespace string, cli kubernetes.Interface) error {
+	for _, service := range services {
+
+		if service.TlsCredentials != "" {
+			_, err := cli.CoreV1().Secrets(namespace).Get(service.TlsCredentials, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("SslProfile %s for service %s does not exist in this cluster", service.TlsCredentials, service.Address)
+			}
+		}
+
+		if service.TlsCertAuthority != "" {
+			_, err := cli.CoreV1().Secrets(namespace).Get(service.TlsCertAuthority, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("SslProfile %s for service %s does not exist in this cluster", service.TlsCertAuthority, service.Address)
+			}
+		}
+	}
+	return nil
+}
+
+func includeSslProfile(sslProfile qdr.SslProfile, tlsManager TlsManagerInterface) error {
```

Comment:
Suggest removing this wrapper func as calling tlsManager.AddSslProfile directly accomplishes same

Reply:
Done!